### DISCUSSION TITLE
Moving/renaming code

### DISCUSSION THREAD

**Discussion by Pandagineer** [bitcraze/discussions#289] [Q&A] [answered]
Hello. I got the "Hello World" example to work. Next, because I want the Bitcraze tutorials to remain untouched, I copied "Hello World" to a new folder, and renamed it. More specifically, on the Desktop I created a new folder "Personal_Projects" and I copied all of the files from "..Crazyflie-firmware/examples/app_hello_world" to it, and then I renamed the "app_hello_world" folder to "Measure_Voltage".

But now when I try to run the Measure_Voltage project (which is just Hello World), it doesn't work. I know this because I changed the "Hello World" string slightly, but the Console is still showing the original "Hello World".

When I make the Measure_Voltage project, I don't see any obvious errors. But perhaps I'm not looking closely enough.

Is there something about moving a project that breaks it? Please advise. Thanks!

**Comment by krichardsson** [comment, marked as answer]
Hi!

There is nothing special with moving a project and it should be fine to move it outside the Crazyflie firmware file tree.

I did a test locally and this is what I did:
1. Copy `examples/app_hello_world` to `/home/kristoffer/code/temp/Measure_Voltage`
2. Clean up of the new folder
  2.1 Remove the `build` directory
  2.2 Remove all `.o` and `.o.cmd` files
3. Rename/change stuff
  3.1 `src/hello_world.c` ==> `src/measure_voltage.c`
  3.2 Updated the `DEBUG_PRINT()` to print something else than "hello world"
4. Update build
  4.1 Update `src/Kbuild` to `obj-y += measure_voltage.o`
  4.2 Update `CRAZYFLIE_BASE` in `Makefile` to point at your crazyflie firmware repo (`CRAZYFLIE_BASE := /home/kristoffer/code/bitcraze/crazyflie-firmware` in my case)
5. `make`
6. `make cload`

You might want to be careful with paths that contains spaces. It can probably be handled but I find that it is easier to avoid. 

I think your problems will be solved by step 2. It seems as `make clean` is not cleaning out everything it should, see bitcraze/crazyflie-firmware#973

**Reply by Pandagineer** [reply]
Hello. I ran into some errors. Here is the last chunk of my make output:

make[2]: Leaving directory '/home/bitcraze/Desktop/Personal_Projects/Measure_Voltage/build'
make[1]: *** [tools/kbuild/Makefile.kbuild:150: sub-make] Error 2
make[1]: Leaving directory '/home/bitcraze/projects/crazyflie-firmware'
make: *** [/home/bitcraze/Desktop/projects/crazyflie-firmware/tools/make/oot.mk:23: all] Error 2
[Make output.txt](https://github.com/bitcraze/discussions/files/9739184/Make.output.txt)

Question: in step 4.2, is this intended to point back to the original crazyflie-firmware folder? Can you explain why our new folder needs to point to a second folder? Are there files in the original folder which we did not copy and the new directory still relies on them?

My entire make output is attached.

**Comment by krichardsson** [comment]
Yes, CRAZYFLIE_BASE should point at the directory with the Crazyflie-firmware source code.

When you build an app you actually build the full firmware + your app code, and the resulting binary is what you flash to the Crazyflie. This is why we need a link to the full firmware source code.

Unfortunately your make output file seems to be empty. 

There is also some documentation for [out of tree builds](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/development/oot/) that might give some insights?

**Reply by Pandagineer** [reply]
I will try to attach again.
[Make Output.txt](https://github.com/bitcraze/discussions/files/9745563/Make.Output.txt)

**Reply by krichardsson** [reply]
It looks like Measure_Voltage.o maybe is missing in your `Kbuild` file?

**Comment by Pandagineer** [comment]
Success! Yes, I realized there are two Kbuild files. I had changed both to contain Measure_Voltage.o, but one of them needs to retain src/.
Thanks!

**Reply by krichardsson** [reply]
Good stuff!
Please also click "mark as answer" to help other users find the solution. Thanks!
